modules/developers-guide/pages/graphql-oriented-using.adoc

```diff
@@ -0,0 +1,1062 @@
+
+include::graphql.adoc[tag=UsingPostman]
+
+== Creating or dropping schema
+
+In order to use the GraphQL API, you must xref:graphql-oriented-using.adoc##_deploying_schema[deploy schema]
+that defines the types, mutations, and queries.
+However, a keyspace, a container that stores the data, must first be created.
+
+// tag::CreateKS[]
+=== Creating a keyspace
+
+Before you can start using the GraphQL API, you must first create a Cassandra
+keyspace in your database.
+
+// LLP: need to check how to write this - it's migration...If you are connecting to a Cassandra database with existing schema, you can skip this step.
+
+Inside the GraphQL playground, navigate to
+http://localhost:8080/graphql-schema[http://localhost:8080/graphql-schema]
+and create a keyspace by executing the following mutation:
+
+[source, plaintext]
+----
+include::example$graphql/createKeyspace.graphql[]
+----
+For each keyspace created in your Cassandra schema, a new path is created under
+the `graphql-path` root (default is: `/graphql`). For example, the mutation just
+executed creates a path `/graphql/library` for the `library` keyspace when
+Cassandra creates the keyspace.
+
+Add the auth token to the HTTP Headers box in the lower lefthand corner:
+
+[source, plaintext]
+----
+{
+  "X-Cassandra-Token":"bff43799-4682-4375-99e8-23c8a9d0f304"
+}
+----
+
+[IMPORTANT]
+====
+Notice that the key for this JSON token is different than the value that the
+generate token has. It is `X-Cassandra-Token`, not `auth-token`.
+====
+
+Now run the mutation to create the keyspace. You should see a return value of:
+
+[source, plaintext]
+----
+include::example$result/gql_createKeyspace.result[]
+----
+// end::CreateKS[]
+
+// tag::CreateUDT[]
+=== Creating a user-defined type (UDT)
+
+User-defined types (UDTs) can be created as GraphQL object types.
+UDTs are optional, but if you wish to use a UDT in another object type definition,
+you'll want to create the UDT first.
+Once created, you can include the type in the schema deployed via the GraphQL playground.
+Here are two examples that create a UDT called `Address` that includes a street, city, state, and zipcode,
+and a UDT called `Review` that includes a book title, comment, rating and review data.
+
+[tabs]
+====
+Address UDT::
++
+--
+[source,shell]
+----
+include::example$graphql/1createUDTAddress.graphql[]
+----
+--
+
+Review UDT::
++
+--
+[source,shell]
+----
+include::example$graphql/1createUDTReview.graphql[]
+----
+--
+====
+
+// USE asciidoc callouts to explain? a la https://docs.couchbase.com/home/contribute/code-blocks.html#callouts
+These types have some Cassandra-specific features, in addition to the typical format
+of a GraphQL type.
+To denote that the type will be stored as a UDT in Cassandra, `@cql_entity(target: UDT)`
+is added to the initial line.
+
+To avoid duplicating the type as both an input type and output type, `@cql_input` is
+added to make the type an input type.
```

Comment:
nit: to generate an `AddressInput` type with the same fields as address in the final GraphQL schema.

Reply:
changed